src/kafka-util/src/client.rs

```diff
@@ -90,3 +91,94 @@ pub fn get_partitions<C: ClientContext>(
+    current_level: tracing::Level,
+    config: &mut rdkafka::config::ClientConfig,
+) {
+    if current_level >= tracing::Level::DEBUG {
+        tracing::debug!(target: "librdkafka", "Enabling 'debug' for rdkafka");
+        config.set("debug", "all");
+    }
+}
+
+/// Return the rdkafka log level based on the tracing level for `librdkafka`
+fn log_level_from_tracing(tracing_level: Level) -> RDKafkaLogLevel {
+    let level = if tracing_level >= Level::DEBUG {
+        RDKafkaLogLevel::Debug
+    } else if tracing_level >= Level::INFO {
+        RDKafkaLogLevel::Info
+    } else if tracing_level >= Level::WARN {
+        RDKafkaLogLevel::Warning
+    } else {
+        RDKafkaLogLevel::Error
+    };
+
+    // WARNING WARNING WARNING
+    //
+    // For whatever reason, if you change this `target` to something else, this
+    // log line might break. I (guswynn) did some extensive investigation with
+    // the tracing folks, and we learned that this edge case only happens with
+    // 1. a different target
+    // 2. only this file (so far as I can tell)
+    // 3. only in certain subscriber combinations
+    // 4. only if the `tracing-log` feature is on.
+    //
+    // Our conclusion was that one of our dependencies is doing something
+    // problematic with `log`.
+    //
+    // For now, this works, and prints a nice log line exactly when we want it.
+    //
+    // TODO(guswynn): when we can remove `tracing-log`, remove this warning
+    tracing::debug!(target: "librdkafka", level = ?level, "Determined log level for librdkafka");
+
+    level
+}
+
+/// A simpler version of [`create_new_client_config`] that defaults
+/// the `log_level` to `INFO` and should only be used in tests.
+#[allow(clippy::disallowed_methods)]
+pub fn create_new_client_config_simple() -> ClientConfig {
+    let mut config = ClientConfig::new();
+    config.set_log_level(log_level_from_tracing(tracing::Level::INFO));
+    config
+}
+
+/// Build a new [`rdkafka`] [`ClientConfig`] with its `log_level` set correctly
+/// based on the passed through [`tracing::Level`]. This level should be
+/// determined for `target: "librdkafka"`.
+#[allow(clippy::disallowed_methods)]
+pub fn create_new_client_config(tracing_level: Level) -> ClientConfig {
+    let mut config = ClientConfig::new();
+    config.set_log_level(log_level_from_tracing(tracing_level));
+    config
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_log_level_from_tracing() {
+        assert!(matches!(
+            log_level_from_tracing(Level::TRACE),
+            RDKafkaLogLevel::Debug
+        ));
+        assert!(matches!(
+            log_level_from_tracing(Level::DEBUG),
+            RDKafkaLogLevel::Debug
+        ));
+        assert!(matches!(
+            log_level_from_tracing(Level::INFO),
+            RDKafkaLogLevel::Info
+        ));
+        assert!(matches!(
+            log_level_from_tracing(Level::ERROR),
+            RDKafkaLogLevel::Error
+        ));
+    }
+}
```

Comment:
I'm not convinced this test is worth it. It's just double bookkeeping on `log_level_from_tracing`.

Reply:
you can remove if you want!

Reply:
you can remove if you want!